docs/rfc/block-registration.md

```diff
@@ -0,0 +1,369 @@
+
+* Gives a name to the block type.
+* Defines some important metadata about the registered block type (title, category, icon, description, keywords).
+* Defines the attributes of your block type.
+* Links to the editor implementation of your block type, the save function and any other context-aware property.
+
+**Example:**
+
+
+```json
+{
+	"name": "my-plugin/notice",
+	"title": "Notice",
+	"category": "common",
+	"icon": "star",
+	"description": "Shows warning, error or success notices  ...",
+	"keywords": [ "alert", "message" ],
+	"attributes": {  /* Block attributes definition */ },
+	"edit": "blocks/notice-edit.js",
+	"save": "blocks/notice-save.js",
+}
+```
+
+## Block API
+
+This section describes all the properties that can be added to the `block.json` file to define the behavior and metadata of block types.
+
+### Name
+
+* Type: string
+* Required
+* Localized: No
+* Property: `name`
+
+```json
+{ "name": "core/heading" }
+```
+
+The name for a block is a unique string that identifies a block. Names have to be structured as `namespace/block-name`, where namespace is the name of your plugin or theme.
+
+**Note:** A block name can only contain lowercase alphanumeric characters, dashes, and at most one forward slash to designate the plugin-unique namespace prefix. It must begin with a letter.
+
+**Note:** This name is used on the comment delimiters as `<!-- wp:my-plugin/book -->`. Block types in the `core` namespace do not include a namespace when serialized.
+
+**Important Note:**
+
+Other block properties that point to JavaScript files use this identifier to provide values of the given properties.
+
+For instance, to define an SVG icon in a `block.js` file, you should attach it to the blocks global variable like so:
+
+```js
+blocks[ 'core/heading' ].icon = // SVG element of the icon. 
+```
+
+### Title
+
+* Type: string
+* Required
+* Localized: Yes
+* Property: `title`
+
+```json
+{ "title": "Heading" }
+```
+
+This is the display title for your block, which can be translated with our translation functions. The block inserter will show this name.
+
+### Category
+
+* Type: string
+* Required
+* Localized: No
+* Property: `category`
+
+```json
+{ "category": "common" }
+```
+
+Blocks are grouped into categories to help users browse and discover them.
+
+The core provided categories are:
+
+* common
+* formatting
+* layout
+* widgets
+* embed
+
+Plugins and Themes can also register [custom block categories](https://wordpress.org/gutenberg/handbook/designers-developers/developers/filters/block-filters/#managing-block-categories).
+
+An implementation should expect and tolerate unknown categories, providing some reasonable fallback behavior (e.g. a "common" category).
+
+### Icon
+
+* Type: object
+* Optional
+* Localized: No
+* Property: `icon`
+
+```json
+{
+	"slug": "star", // Dashicon slug, serve as a fallback if non-js contexts. 
+	"src": "./my-file.js", // Path to a JavaScript file containing the block's icon property
```

Comment:
For me these are not meant to be loaded by the browser directly, instead a smart "bundler" should be able to traverse it and include the dependencies, like webpack does for "css/js/younameit" imports.

Reply:
> For me these JSON files are not meant to be loaded by the browser directly, instead a smart "bundler" should be able to traverse it and include the dependencies, like webpack does for "css/js/younameit" imports.

Is this worth stating explicitly in the RFC?

Reply:
You can't bundle JSON directly into the browser. I think this is an implementation detail and initially, it might be even not a case for the WordPress client code as we will expose those definitions through PHP anyway.

It's more a future consideration for those consumers who plan to use core blocks without WordPress as a backend. I think it's fine to leave this issue to be solved for themselves. On our end we will have to process those JSON files and replicated servers-side exposure for our unit tests.